### DISCUSSION TITLE
Myanmar (Burmese) Language

### DISCUSSION THREAD

**Discussion by MgThein199** [RHVoice/RHVoice#564] [Ideas]
Please support **Myanmar (Burmese)** Language. I'll provide all data you needed.

**Comment by louderpages** [comment]
Hello.  May I ask a question about your language?

The writing has no punctuation within sentences, and no breaks between words, am I correct?

Our system needs these in order to learn and then pronounce speech with some kind of prosody - that is the rhythm with changing speed and loudness.  This seems an intractable problem for our technology, which does not understand the meaning of what is being said, and so will not know how to group syllables in to words, and place pauses between words when appropriate.

Or, am I missing something?

Regards,
- LP

**Comment by MgThein199** [comment]
Thanks for your interest.

Right.
1. Spelling sounds and pronunciations are different in Burmese(Often but not always). 

2. How to group syllables into words?
There is no breaks between words.
But we can break only one way.
In the word "ဖတ်စာအုပ်"(means "textbook"), we can break "ဖတ်,စာ , အုပ်". But we can't do other breaks (e.g "ဖ, တ်, စာ, အု,ပ်" has no pronunciation and no meaning). 
Burmese has no linking words/sounds. 

3. Speed and loudness
In Burmese, there are **85 different sounds** (consonants, consonant cluster, vowels, diphthongs). I believe those sounds can make pronunciation correct if we use all sounds in the project.

4. Pauses
In Burmese spaces can be found between phrases. 
We use pause the place of "။" (like "full stop").
For the space, pauses are short.


To group words and to pronounce correctly, could we add a **word-pronunciation** dictionary (lists) of the words(or phrases) and pronunciation scripts (IPA or X-SAMPA ) in the project? 

[85_Burmese_Sounds.pdf](https://github.com/RHVoice/RHVoice/files/8714399/85_Burmese_Sounds.pdf)
[Example_word_pronunciation_list.pdf](https://github.com/RHVoice/RHVoice/files/8714400/Example_word_pronunciation_list.pdf)

Regards,

**Comment by louderpages** [comment]
There are a lot of sounds, yes, which means a lot of training recording.

But I want to understand what you mean exactly when you say:
> There is no breaks between words.
> But we can break only one way.
> 

How is the correct way to break known?  Is it a purely phonetic thing. For example. words cannot start or end with a certain sequence of syllables?

Or is it because human intelligence just knows which breaks give meaning to the sentence?   For example, the following sentence in English has no spaces between words.

Therearenospaceshere.

I know many thousands of words, I know their meaning and I know the grammar.  I know that “space shere” makes no sense, but “spaces here” does.  So I know this says, “There are no spaces here."

Now, because I am a native speaker, my brain just automatically knows that there is only one way to break this up into words.  Just like my brain does so when listening to speech, where there are no pauses between words. It is subconscious.  

Is that what you do with Burmese?  It is impossible to automate this in RHVoice.

How do sighted children learn to read Burmese? Are they given books with marks breaking the text into words?

By the way, how about Sao Mai TTS?

Regards
- LP

**Reply by MgThein199** [reply]
Thanks Sir,
- Are the children given books with marks breaking the text into words?
- No.
- How do sighted children learn to read Burmese? 
- We make children to be familiar with the words. They have to recite many words (according to their ages).
- e.g.
Consonants+ ု, ူ, ူး 
ကု, ကူ, ကူး
ခု, ခူ, ခူး
etc.
They have to read many words which can combine with "ု, ူ, ူး". 
________
This document is written to explain breaking Burmese text into words in layman's terms.
**Please give me any suggestions or ideas or advice.**
[To_Break_Continuous_Burmese_Word.pdf](https://github.com/RHVoice/RHVoice/files/8838522/To_Break_Continuous_Burmese_Word.pdf)
Unicode Chart
[U1000.pdf](https://github.com/RHVoice/RHVoice/files/8838524/U1000.pdf)

Regards,
MgThein

**Reply by louderpages** [reply]
Dear  @MgThein199,

My apologies for the delay in responding.  I have looked at the pdf you linked to about word-breaks, and I can see that the meaning of the phrase does not need to be known. The words can be split by syntax alone.

And, the vowels on tones can always be detected from the writing, is that correct?

Lots of sounds, lots of rules, lots of recording, lots of work. Smile.

Can you tell me what the need is?  What is wrong with existing voices?

If you wish to contact me privately, please email me: inq at louder pages dot org.

Kind regard,
- Mark

**Comment by zstanecic** [comment]
hi @louderpages!
I have seen what's the problem with this language:
This language really uses script, in which you write all words in continuo, i.e. really without spaces, respectively.
So,forexample,youwriteallwordsandpunctuationinthismannerandyourprocessitinyourbrain.
Sao mai tts exists, but it is not so responsive when it comes to work with the language.

**Comment by louderpages** [comment]
Yes, @zstanecic but I think you did not see the previous discussion about this exact issue.
